internal/cmd/skupper/connector/kube/connector_generate.go

```diff
@@ -0,0 +1,250 @@
+package kube
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/skupperproject/skupper/internal/cmd/skupper/common"
+	"github.com/skupperproject/skupper/internal/cmd/skupper/common/utils"
+
+	"github.com/skupperproject/skupper/internal/kube/client"
+	"github.com/skupperproject/skupper/pkg/apis/skupper/v2alpha1"
+	skupperv2alpha1 "github.com/skupperproject/skupper/pkg/generated/client/clientset/versioned/typed/skupper/v2alpha1"
+	pkgUtils "github.com/skupperproject/skupper/pkg/utils"
+	"github.com/skupperproject/skupper/pkg/utils/validator"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type CmdConnectorGenerate struct {
+	client              skupperv2alpha1.SkupperV2alpha1Interface
+	CobraCmd            *cobra.Command
+	Flags               *common.CommandConnectorGenerateFlags
+	namespace           string
+	name                string
+	port                int
+	output              string
+	host                string
+	selector            string
+	tlsCredentials      string
+	routingKey          string
+	connectorType       string
+	includeNotReadyPods bool
+	KubeClient          kubernetes.Interface
+}
+
+func NewCmdConnectorGenerate() *CmdConnectorGenerate {
+
+	return &CmdConnectorGenerate{}
+}
+
+func (cmd *CmdConnectorGenerate) NewClient(cobraCommand *cobra.Command, args []string) {
+	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), cobraCommand.Flag("kubeconfig").Value.String())
```

Comment:
The generate commands need to work in a environment where there is no kubernetes installed because their main purpose is to just generate a resource. Hence, all the checks that depends on existing resources (like checking if a site or workload exists) don't apply to the `generate` commands. The only validations that have to be maintained are the ones that check if the values introduced by the user are incorrect.

Reply:
thanks, I wasn't sure about that, will remove.

Reply:
@nluaces for workload option once you remove getting the values from k8 I need to put something in the selector field.
```
apiVersion: skupper.io/v2alpha1
kind: Connector
metadata:
  name: backup
  namespace: default
spec:
  port: 8080
  routingKey: backup
  selector: ?????
  type: tcp
```
I'm verifying the user doesn't specify more than one host/selector/workload 
but if user specifies workload what do you think I should put in the selector field:   
options:
   1.  just put in what user specified   <deployment/backend>
 
   2. specify where I get the selector from    
        if deployment:
        selector: <deployment.Spec,Selector.MatchLabels>
   3. ????

Reply:
@lynnemorrison if there is no kubernetes environment we can't obtain the selector of a given workload, because this has to work just as a generator of resources.

My suggestion is that if we don't have a selector, to do not specify the selector field in the resource.

cc: @ssorj

Reply:
I've removed all kubernetes calls from the generate commands.